modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -220,25 +214,15 @@ else if (matches > 1)
         return extensionRegistry.getConfigurationProvider(configurationProviderName);
     }
 
-    private boolean attemptToCreateImplicitConfiguration(ExtensionModel extensionModel, MuleEvent muleEvent)
+    private void createImplicitConfiguration(ExtensionModel extensionModel, MuleEvent muleEvent)
     {
         synchronized (extensionModel)
         {
             //check that another thread didn't beat us to create the instance
-            if (!extensionRegistry.getConfigurationProviders(extensionModel).isEmpty())
+            if (extensionRegistry.getConfigurationProviders(extensionModel).isEmpty())
```

Comment:
but if you already have configuration providers registered, by do you call registerConfigurationPRovder() again? you should use the one that exists.

What I'm questioning is that the if seems to be missing a `!`